### PR TITLE
toFloat() monotonicity fixed

### LINKED ISSUE #6248: Confused by toFloat64() comparison
The id column in myTable is stored as string. However, it is a valid number. In the query below, one row is returned and the compare_indicator is 1.
``` 
select id, toFloat64(id) = 439385272.0 as compare_indicator
from myTable
where id = '439385272' 
```
In the query below, nothing is returned. 
```
 select id, toFloat64(id) = 439385272.0 as compare_indicator
 from myTable
 where toFloat64(id) = 439385272.0 
```
Even given a value range, the query below returns nothing.
```
 select id, toFloat64(id) = 439385272.0 as compare_indicator
 from myTable
 where toFloat64(id) between 439385262.0 and 439385282.0
```
Am I missing something? Or is it a possible bug?

### PR DESCRIPTION
**Pull request by dimarub2000**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Fixes #6248

### CODE REVIEW

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Not enough.

**Comment by alexey-milovidov** [comment]
Need to add a test:
- for the issue;
- to check that the index by integer or decimal continue to work if you convert toFloat64.

**Comment by dimarub2000** [comment]
Tests will be added soon.

**Comment by alexey-milovidov** [comment]
Still not ready.

File: dbms/src/Functions/FunctionsConversion.h [hunk 1]
```diff
@@ -1366,8 +1366,8 @@ using FunctionToInt8 = FunctionConvert<DataTypeInt8, NameToInt8, ToIntMonotonici
 using FunctionToInt16 = FunctionConvert<DataTypeInt16, NameToInt16, ToIntMonotonicity<Int16>>;
 using FunctionToInt32 = FunctionConvert<DataTypeInt32, NameToInt32, ToIntMonotonicity<Int32>>;
 using FunctionToInt64 = FunctionConvert<DataTypeInt64, NameToInt64, ToIntMonotonicity<Int64>>;
-using FunctionToFloat32 = FunctionConvert<DataTypeFloat32, NameToFloat32, PositiveMonotonicity>;
-using FunctionToFloat64 = FunctionConvert<DataTypeFloat64, NameToFloat64, PositiveMonotonicity>;
+using FunctionToFloat32 = FunctionConvert<DataTypeFloat32, NameToFloat32, ToIntMonotonicity<Float32>>;
```
Comment (alexey-milovidov) [inline dbms/src/Functions/FunctionsConversion.h]: But now the template name is idiosyncratic.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**